### PR TITLE
Fix/add o all

### LINKED ISSUE #643: -o all option needs to be added to matrix*ncep
**Describe the bug**
The -o all option was supposed to be brought to the matrix files, but was not put in matrix*ncep or unintentionally dropped. 

**To Reproduce**
Run matrix_cmake_ncep 
there will be no netcdf output as @aliabdolali discovered. 

**Expected behavior**
All output should be created.

### PR DESCRIPTION
**Pull request by aliabdolali**
# Pull Request Summary
add `-o all` to matrix*ncep

## Description
The -o all option was supposed to be brought to the matrix files, but was not put in matrix*ncep or unintentionally dropped.

Please also include the following information: 
* Add any suggestions for a reviewer @JessicaMeixner-NOAA 

### Issue(s) addressed

* Please list any issues associated with this PR, including those the PR will fix/close. For example:  
- fixes #643
- closes #643

### Commit Message
add -o all option to matrix_cmake_ncep

### Check list  

- [x] Branch is up to date with the authoritative repository (NOAA-EMC) develop branch. 
- [x] Checked the [checklist for a developer submitting to develop](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-a-developer-submitting-to-develop). 
- [n/a] If a version number update is required, checked the [updating version number](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-updating-version-number) checklist. 
- [n/a] If a new feature was added, a regression test for testing the new feature is added. 
 
### Testing

* How were these changes tested? to be updated
* Are the changes covered by regression tests? n/a
* Have the matrix regression tests been run (if yes, please note HPC and compiler)? Hera/intel
* Please provide the summary output of matrix.comp (_matrix.Diff.txt_, _matrixCompFull.txt_ and _matrixCompSummary.txt_):  to be updated
* Please indicate the expected changes in the regression test output ([Note the known list of non-identical tests](https://github.com/NOAA-EMC/WW3/wiki/How-to-use-matrix.comp-to-compare-regtests-with-master#4-look-at-results)).
* Please provide the summary output of matrix.comp (_matrix.Diff.txt_, _matrixCompFull.txt_ and _matrixCompSummary.txt_):
to be updated

### CODE REVIEW

File: regtests/bin/matrix_cmake_ncep [hunk 1]
```diff
@@ -165,10 +165,11 @@ main_dir="`cd $main_dir 1>/dev/null 2>&1 && pwd`"
 # Compile option
   if [ "$cmplOption" = 'y' ]
   then
-     opt="-c $cmplr -S -T"
+     opt="-o all -c $cmplr -S -T"
```
Comment (JessicaMeixner-NOAA) [inline regtests/bin/matrix_cmake_ncep:168]: Should this also be added to matrix_ncep just in case someone runs that in the meantime?

Comment (aliabdolali) [reply regtests/bin/matrix_cmake_ncep:168]: it is already there https://github.com/NOAA-EMC/WW3/blob/develop/regtests/bin/matrix_ncep#L164-L170
The way i noticed it was I first ran with the cmake and then ran with the old one.

Comment (JessicaMeixner-NOAA) [reply regtests/bin/matrix_cmake_ncep:168]: Thanks!

**Review by JessicaMeixner-NOAA — APPROVED** [review]

**Comment by JessicaMeixner-NOAA** [comment]
@aliabdolali I ran a similar branch and the diffs I got with the previous develop (ignoring the 0 diff entries) are: 
```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR2_UQ_MPI_d2                     (8 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (6 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (8 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (8 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (12 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (8 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (9 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (5 files differ)
ww3_tp2.14/./work_OASACM2                     (2 files differ)
ww3_tp2.14/./work_OASACM                     (2 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (2 files differ)
ww3_ufs1.3/./work_a                     (1 files differ)
```
I think it's okay to go ahead and merge this branch.

**Comment by aliabdolali** [comment]
> add -o all option to matrix_cmake_ncep

My requests are done with expected changes. Thanks for the parallel test.

**Pull request merged**